piet/src/color.rs

```diff
@@ -7,7 +7,7 @@ use std::fmt::{Debug, Formatter};
 /// Currently this is only a 32 bit RGBA value, but it will likely
 /// extend to some form of wide-gamut colorspace, and in the meantime
 /// is useful for giving programs proper type.
-#[derive(Clone, PartialEq)]
+#[derive(Clone, PartialEq, Hash)]
```

Comment:
This is sticky. The reason we don't implement Hash here is that future extensions of Color might include floating point values. I'm wrestling with the question because of course there's no compatibility guarantee, and we can just take away Hash if and when we get there, or alternatively decide that we can hash the bit patterns.

But it would ease my mind slightly to use the as_rgba_u32 value instead as your hash key.